### PR TITLE
cut_mix example: tf.cast label from int to float

### PR DESCRIPTION
**Pull request by johnypark**
Hello,

I noticed that the current example for cut_mix does not work, because

the dtype is int for the label:
```
---------------------------------------------------------------------------
InvalidArgumentError                      Traceback (most recent call last)
[<ipython-input-80-2593aa7caea0>](https://localhost:8080/#) in <module>()
      1 (images, labels), _ = tf.keras.datasets.cifar10.load_data()
      2 cutmix = keras_cv.layers.preprocessing.cut_mix.CutMix(10)
----> 3 output = cutmix({'images': images, 'labels': labels})

1 frames
[/usr/local/lib/python3.7/dist-packages/keras/utils/traceback_utils.py](https://localhost:8080/#) in error_handler(*args, **kwargs)
     65     except Exception as e:  # pylint: disable=broad-except
     66       filtered_tb = _process_traceback_frames(e.__traceback__)
---> 67       raise e.with_traceback(filtered_tb) from None
     68     finally:
     69       del filtered_tb

[/usr/local/lib/python3.7/dist-packages/tensorflow/python/framework/ops.py](https://localhost:8080/#) in raise_from_not_ok_status(e, name)
   7162 def raise_from_not_ok_status(e, name):
   7163   e.message += (" name: " + name if name is not None else "")
-> 7164   raise core._status_to_exception(e) from None  # pylint: disable=protected-access
   7165 
   7166 

InvalidArgumentError: Exception encountered when calling layer "cut_mix_4" (type CutMix).

cannot compute Mul as input #1(zero-based) was expected to be a float tensor but is a uint8 tensor [Op:Mul]

Call arguments received by layer "cut_mix_4" (type CutMix):
  • inputs={'images': 'tf.Tensor(shape=(50000, 32, 32, 3), dtype=float32)', 'labels': 'tf.Tensor(shape=(50000, 1), dtype=uint8)'}
  • training=True
```

Changing the dtype of ```labels``` to ```tf.cast(labels, dtype = tf.float32)``` solves the issue.

Example notebook:
https://colab.research.google.com/drive/1k-IrSJF-DlP_o27ohr7GvYsXliGRifTF?usp=sharing

### CODE REVIEW

**Comment by LukeWood** [comment]
Hey Johnny,

this is a great bug report in that it solidifies to me we need better communication in our error messages.  CutMix modifies the labels as if they are a continuous space.  This requires them to be one hot encoded labels of type float32.  I'll go ahead and do some writing to clarify this.

I'll open a PR to add a better error reporting message for this.

Thanks for the report, this feedback is highly valuable

**Comment by LukeWood** [comment]
The docstring is actually fully wrong now that I look at it.  We need to one hot encode the labels.  Can you update your PR to include `one_hot` in the sample code?
 
We should do the same for mix_up too.

**Comment by johnypark** [comment]
Hi @LukeWood, sure thing! Let me change it and get back soon.

**Comment by johnypark** [comment]
@LukeWood,

I made changes to the cut_mix example to implement tf.one_hot in the label. 

I noticed that keras_cv has a demo for cut_mix as well, and tested it also not working. 

So I made some changes so that the cut_mix demo could work. 

While doing so, I realized that the output of the cut_mix is a dictionary with keys {images, labels}, instead of (images, labels) format as other preprocessing layers. 

So I made changes here https://github.com/keras-team/keras-cv/pull/445/commits/1d1a91d895eddbfe62a14a6119fa7bbc9b2aaf40 to change the output from dictionary to (images, labels). Please let me know whether this changes makes sense.

**Comment by LukeWood** [comment]
> While doing so, I realized an [issue](https://github.com/keras-team/keras-cv/issues/447) that the output of the cut_mix is a dictionary with keys {images, labels}, instead of (images, labels) format as other preprocessing layers.

Actually, the other preprocessing labels do also output a dictionary.  Everything does.  This is done to support "bounding_boxes" targets, and "segmentation_map" targets.

File: keras_cv/layers/preprocessing/cut_mix.py [hunk 1]
```diff
@@ -33,7 +33,8 @@ class CutMix(tf.keras.__internal__.layers.BaseImageAugmentationLayer):
     ```python
     (images, labels), _ = tf.keras.datasets.cifar10.load_data()
     cutmix = keras_cv.layers.preprocessing.cut_mix.CutMix(10)
-    output = cutmix({'images': images, 'labels': labels})
+    images, labels =  images[:32], labels[:32]
```
Comment (LukeWood) [inline keras_cv/layers/preprocessing/cut_mix.py:36]: can you one hot in an extra line and add the same code to the mixup demo/docstring?

So how about:
```
(images, labels), _ = tf.keras.datasets.cifar10.load_data()
labels = tf.one_hot(labels.squeeze(), 10)
cutmix = keras_cv.layers.preprocessing.cut_mix.CutMix(10)
cutmix({"images": images[:32], "labels": labels[:32]})
```

File: keras_cv/layers/preprocessing/cut_mix.py [hunk 2]
```diff
@@ -59,9 +60,9 @@ def _batch_augment(self, inputs):
                 f"Got: inputs = {inputs}"
             )
         images, labels = self._update_labels(*self._cutmix(images, labels))
-        inputs["images"] = images
-        inputs["labels"] = labels
-        return inputs
+        #inputs["images"] = images
```
Comment (LukeWood) [inline keras_cv/layers/preprocessing/cut_mix.py:63]: Let's revert these changes and keep the PR localized to the docstring update.  Thanks!

**Comment by LukeWood** [comment]
Can you make the docstring changes to mixup too?

File: keras_cv/layers/preprocessing/cut_mix.py (hunk 1 above)
Comment (johnypark) [reply keras_cv/layers/preprocessing/cut_mix.py:36]: > how about:
> 
> ```
> (images, labels), _ = tf.keras.datasets.cifar10.load_data()
> labels = tf.one_hot(labels.squeeze(), 10)
> 
> cutmix = keras_cv.layers.preprocessing.cut_mix.CutMix(10)
> cutmix({"images": images[:32], "labels": labels[:32]})
> ```
> 
> (Please include the newline too!)

I think that works perfectly. Let me make that change.

File: keras_cv/layers/preprocessing/cut_mix.py (hunk 2 above)
Comment (johnypark) [reply keras_cv/layers/preprocessing/cut_mix.py]: > 
Reverted changes.

**Comment by johnypark** [comment]
> > While doing so, I realized an [issue](https://github.com/keras-team/keras-cv/issues/447) that the output of the cut_mix is a dictionary with keys {images, labels}, instead of (images, labels) format as other preprocessing layers.
> 
> Actually, the other preprocessing labels do also output a dictionary. Everything does. This is done to support "bounding_boxes" targets, and "segmentation_map" targets.

I see. So that was intended for bounding boxes and segmentation maps. Great to know! Thank you.

**Comment by LukeWood** [comment]
Sorry Johnny I forgot the output = in the last line.  Add that and good to go!

**Review by LukeWood — APPROVED** [review]

**Comment by johnypark** [comment]
@LukeWood, please let me know if anything else needs to be changed. Thank you.

**Comment by johnypark** [comment]
> Sorry Johnny I forgot the output = in the last line. Add that and good to go!

done!

**Review by LukeWood — APPROVED** [review]

**Comment by LukeWood** [comment]
Thanks @johnypark for the contribution!
